system/core/document.php

```diff
@@ -499,26 +500,27 @@ function handleImageUpload($name, $fileinfo) {
 	}
 
 	class HTMLTable extends DOMWrap\Element {
+		/** @var HTMLTableRow */
 		private $currentRow;
 
 		function __construct() {
 			parent::__construct('table');
 		}
 
 		function addHeaderRow() {
-			$currentRow = new HTMLTableRow('th');
-			$this->appendChild($currentRow);
-			return $currentRow;
+			$this->currentRow = new HTMLTableRow('th');
+			$this->appendChild($this->currentRow);
+			return $this->currentRow;
 		}
 
 		function addRow() {
-			$currentRow = new HTMLTableRow('td');
-			$this->appendChild($currentRow);
-			return $currentRow;
+			$this->currentRow = new HTMLTableRow('td');
+			$this->appendChild($this->currentRow);
+			return $this->currentRow;
 		}
 
 		function addCell($contents = null) {
-			return $currentRow->addCell($contents);
+			return $this->currentRow->addCell($contents);
```

Comment:
Did this code actually work before? Or is this really a bugfix?

Reply:
Looking at the code, it could not have worked properly; `$currentRow` was never defined.